js_modules/dagster-ui/packages/ui-core/src/assets/AssetRecentUpdatesTrend.tsx

```diff
@@ -13,32 +13,63 @@ import {Link} from 'react-router-dom';
 import styled from 'styled-components';
 
 import {AssetHealthSummary} from './AssetHealthSummary';
-import {useRecentAssetEvents} from './useRecentAssetEvents';
-import {Timestamp} from '../app/time/Timestamp';
-import {AssetHealthFragment} from '../asset-data/types/AssetHealthDataProvider.types';
-import {MaterializationHistoryEventTypeSelector} from '../graphql/types';
 import {
   AssetFailedToMaterializeFragment,
   AssetObservationFragment,
   AssetSuccessfulMaterializationFragment,
 } from './types/useRecentAssetEvents.types';
+import {useRecentAssetEvents} from './useRecentAssetEvents';
+import {useRefreshAtInterval} from '../app/QueryRefresh';
+import {Timestamp} from '../app/time/Timestamp';
+import {AssetLatestInfoFragment} from '../asset-data/types/AssetBaseDataProvider.types';
+import {AssetHealthFragment} from '../asset-data/types/AssetHealthDataProvider.types';
+import {MaterializationHistoryEventTypeSelector} from '../graphql/types';
 
 export const AssetRecentUpdatesTrend = React.memo(({asset}: {asset: AssetHealthFragment}) => {
   // Wait 100ms to avoid querying during fast scrolling of the table
   const shouldQuery = useDelayedState(100);
-  const {materializations, observations, loading} = useRecentAssetEvents(
+  const {
+    materializations: _materializations,
+    observations: _observations,
+    latestInfo,
+    loading,
+    refetch,
+  } = useRecentAssetEvents(
     shouldQuery ? asset.assetKey : undefined,
     5,
     MaterializationHistoryEventTypeSelector.ALL,
   );
 
+  const {materializations, observations} = useMemo(() => {
+    if (!latestInfo?.inProgressRunIds.length && !latestInfo?.unstartedRunIds.length) {
+      return {materializations: _materializations, observations: _observations};
+    }
+    return {
+      materializations: [latestInfo, ..._materializations],
+      observations: [latestInfo, ..._observations],
+    };
+  }, [latestInfo, _materializations, _observations]);
+
+  useRefreshAtInterval({
+    refresh: refetch,
+    intervalMs: 3000,
```

Comment:
This refresh is a little aggressive... cc @neilfulwiler if we had an event stream to listen to I wouldn't need to do this aggressive polling.

Reply:
Hmm yeah I'm torn because this interval is on the per-asset level?

I think that you may actually be able to remove this and get the appropriately-timed-reloading anyway, if the useRecentAssetEvents hook is pulling latestInfo from the asset live data, since that uses the run logs stream to time the refreshes?

(Or if useRecentAssetEvents is not using that to get latest info, I think you could also get these two states you're looking for from the live data hook separately)

Reply:
ah yeah I was planning to show this information somewhere but will omit it for now. I was thinking of avoiding pull all of the AssetGraphLiveQuery for this

Reply:
can asset graph live query be served from event stream? or what else is being derived from that

Reply:
@bengotow I didn't use AssetGraphLiveQuery because it pulls the whole snapshot :\. I think only pulling assetsLatestInfo is cheaper